### PR TITLE
1680 Fix playbook create permissions

### LINKED ISSUE #1680: create playbook cta active even without permissions
#bug primary button to create playbook is active even without permissions (team override scheme).

Api returns a 403 correctly, but button should be disabled and a tooltip shown.

_Issue created from a [Mattermost message](https://community.mattermost.com/_redirect/pl/hgupcr8hujghdx1jpwzghzbjgo) *by @trilopin*._

### PR DESCRIPTION
**Pull request by m-ripper**
## Summary

Fix check if playbook can be created when assigned to multiple teams.
Instead of checking if the user has permissions in any team he's member of, it will now only be checked for the currently selected team.

## Ticket Link

Fixes #1680 

## Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Review by trilopin — APPROVED** [review]
Thanks @m-ripper 

please do not merge this pr before we merge the linked-channel one

**Comment by trilopin** [comment]
I guess `useCanCreatePlaybooksOnAnyTeam` is here since the no-team-sidebar ages. Do you think we can remove that function? @m-ripper @lieut-data

File: webapp/src/components/backstage/runs_page_no_content.tsx [hunk 1]
```diff
@@ -91,37 +87,33 @@ const NoContentPlaybookRunSvgContainer = styled.div`
...
-    const teams = useSelector<GlobalState, Team[]>(getMyTeams);
-    const goToMattermost = () => {
-        navigateToUrl('');
-    };
-    const [playbookExist, setplaybookExist] = useState(false);
+    const teamId = useSelector(getCurrentTeamId);
+    const [playbookExist, setPlaybookExist] = useState(false);
+    const [,,
+        {setSelectedPlaybook},
+    ] = usePlaybooksCrud({team_id: '', per_page: BACKSTAGE_LIST_PER_PAGE});
```
Comment (lieut-data) [inline webapp/src/components/backstage/runs_page_no_content.tsx:94]: nit: I wonder if `usePlaybooksCrud` should eventually be migrated to always returning an object, like `usePlaybooksRouting`?

File: webapp/src/components/rhs/rhs_home.tsx [hunk 2]
```diff
@@ -109,10 +109,7 @@ const ListSection = styled.div`
 const RHSHome = () => {
     const dispatch = useDispatch();
     const currentTeam = useSelector(getCurrentTeam);
```
Comment (lieut-data) [inline webapp/src/components/rhs/rhs_home.tsx:111]: Should we simplify to just `getCurrentTeamId` if we only use the id?

**Review by lieut-data — APPROVED** [review]
Thanks, @m-ripper! Can we look to add e2e coverage for this change?

File: webapp/src/components/backstage/runs_page_no_content.tsx (hunk 1 above)
Comment (m-ripper) [reply webapp/src/components/backstage/runs_page_no_content.tsx:94]: That's a good point. I haven't looked into `usePlaybooksCrud` in-depth but judging by the signature there, there should be a better way and I'll gladly take a look if that can be migrated.

**Pull request merged**